### ISSUE TITLE
Usage with dynamic import()

### ISSUE THREAD

**Issue by selbekk** [entwicklerstube/babel-plugin-root-import#78]
Currently, this plugin does not work with dynamic imports, like this:

```javascript
export default async function showHeavyUi() {
  const ExpensiveUi = await import('~/components/expensive-ui');

  return ExpensiveUi;
}
```

Is there a way to make this work?

I'd love to help create this feature, but I've never touched anything inside a babel plugin. I'd love to help test, document or whatever though!

**Comment by michaelzoidl** [comment]
Hi Kristofer, thats because the babel plugin only handles normal `import` statements, if you write a plugin you have to specify which functions you want to handle.

So in this case we have to add the dynamic import function, usually the existing code should work, you've only to add the dynamic function.

But one thing i don't know is, the dynamic import is a ES proposal, so maybe it requires a sub-plugin, thats something you've to investigate :)

It would be great if you add this feature since it seems very interesting :)

**Comment by PinkaminaDianePie** [comment]
any progress on this or maybe workaround?

**Comment by selbekk** [comment]
The workaround is to use relative paths for now @PinkaminaDianePie :)

I'd love to fix this, but I'd have to learn the basics of ASTs first, and I honestly haven't had time.

**Comment by thetoxicavenger** [comment]
+1 for a dynamic imports solution

**Comment by vihanb** [comment]
This is pretty simple to add, in my form [babel-7-plugin-root-import](https://github.com/vihanb/babel-plugin-root-import) I've added this in dfb33ef51ccd3a83bc93f90c3ad8ee25c6135e9b still waiting to get merged in #93

**Comment by brigand** [comment]
This is supported now.
